openwisp_users/api/serializers.py

```diff
@@ -0,0 +1,434 @@
+            except OrganizationUser.DoesNotExist:
+                pass
+            if org_user:
+                if (
+                    str(org_user_data['organization'].id)
+                    in instance.organizations_dict.keys()
+                ):
+                    if org_user.is_admin != org_user_data.get('is_admin'):
+                        org_user.is_admin = org_user_data['is_admin']
+                        org_user.full_clean()
+                        org_user.save()
+                    else:
+                        org_user.delete()
+            else:
+                org_user_data['user'] = instance
+                org_user_instance = OrganizationUser(**org_user_data)
+                org_user_instance.full_clean()
+                org_user_instance.save()
+
+        instance = self.instance or self.Meta.model(**validated_data)
+        instance.full_clean()
+        return super().update(instance, validated_data)
+
+
+def get_userlist_fields(fields):
+    """
+    Returns the fields for `UserListSerializer`.
+    """
+    fields = list(fields)
+    fields.remove('is_superuser')
+    fields = tuple(fields)
+    return fields
+
+
+class UserListSerializer(SuperUserListSerializer):
+    class Meta:
+        model = User
+        fields = get_userlist_fields(SuperUserListSerializer.Meta.fields[:])
+        read_only_fields = ('last_login', 'date_joined')
+        extra_kwargs = {
+            'email': {'required': True},
+            'password': {'write_only': True, 'style': {'input_type': 'password'}},
+        }
+
+
+def get_userdetail_fields(fields):
+    """
+    Returns the fields for `UserDetailSerializer`.
+    """
+    fields = list(fields)
+    fields.remove('is_superuser')
+    fields.remove('user_permissions')
+    fields = tuple(fields)
+    return fields
+
+
+class UserDetailSerializer(SuperUserDetailSerializer):
+    organization_users = OrganizationUserSerializer(required=False, read_only=True)
+
+    class Meta:
+        model = User
+        fields = get_userdetail_fields(SuperUserDetailSerializer.Meta.fields[:])
+        extra_kwargs = {
+            'last_login': {'read_only': True},
+            'date_joined': {'read_only': True},
+        }
+
+
+class ChangePasswordSerializer(serializers.Serializer):
+    current_password = serializers.CharField(
+        allow_null=True, write_only=True, style={'input_type': 'password'}
+    )
+    new_password = serializers.CharField(
+        required=True, write_only=True, style={'input_type': 'password'}
+    )
+    confirm_password = serializers.CharField(
+        required=True, write_only=True, style={'input_type': 'password'}
+    )
+
+    def validate_confirm_password(self, value):
+        if self.initial_data.get('new_password') != self.initial_data.get(
+            'confirm_password'
+        ):
+            raise serializers.ValidationError(
+                _('New password and Confirm password do not match.')
+            )
+        return value
```

Comment:
@ManishShah120 I think there must be a check that `new_password` or `confirm_password` can't be the same as `current_password`. What do you think?

Reply:
> @ManishShah120 I think there must be a check that `new_password` or `confirm_password` can't be the same as `current_password`. What do you think?

Agreed :+1:·

Reply:
Yes, will add this check ASAP, thnx @codesankalp for pointing this out. :+1:

Reply:
Done adding the check for same old and new password. :smile: